### PR TITLE
Create sync_docs.yaml

### PR DESCRIPTION
**Pull request by a-velasco**
## Issue
Discourse docs are not versioned/in sync with the repository.

## Solution
Added a new workflow that pulls docs from Discourse whenever a branch is pushed to main.

Uses workflow from [data-platform-workflows](https://github.com/canonical/data-platform-workflows/blob/main/.github/workflows/_sync_docs.md) based on [deusebio/discourse-gatekeeper](https://github.com/deusebio/discourse-gatekeeper).

### CODE REVIEW

File: .github/workflows/sync_docs.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# See LICENSE file for licensing details.
+name: Sync docs from Discourse
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 50 00 * * *
+  push:
+    branches:
+      - main
```
[the remark this answers is not in the document]
Comment (a-velasco) [reply .github/workflows/sync_docs.yaml:11]: No, this is strictly disabled.

File: .github/workflows/sync_docs.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+jobs:
+  sync-docs:
+    name: Sync docs from Discourse
+    uses: canonical/data-platform-workflows/.github/workflows/_sync_docs.yaml@v13.3.0
```
[the remark this answers is not in the document]
Comment (taurus-forever) [reply .github/workflows/sync_docs.yaml:16]: Why is renovate not updating dpw here?

File: .github/workflows/sync_docs.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    secrets:
+      discourse_api_username: ${{ secrets.DISCOURSE_API_USERNAME }}
+      discourse_api_key: ${{ secrets.DISCOURSE_API_KEY }}
```
[the remark this answers is not in the document]
Comment (taurus-forever) [reply .github/workflows/sync_docs.yaml:19]: Yes. Should be.

File: .github/workflows/sync_docs.yaml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+name: Sync docs from Discourse
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 50 00 * * *
```
Comment (taurus-forever) [inline .github/workflows/sync_docs.yaml:8]: It is a mongodb-k8s time: https://docs.google.com/spreadsheets/d/1crHqnVtuGHsNIrOYGsfkIiGUA7peJNFNOqMkvTKVvd4/edit#gid=0

PG K8s is 00:30

File: .github/workflows/sync_docs.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (a-velasco) [reply .github/workflows/sync_docs.yaml:11]: Good point! While it is still one-way sync, the cron job should be enough.

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**